docs/core/advanced_tutorials/using-result-handlers.md

```diff
@@ -0,0 +1,140 @@
+def add(x, y=1):
+   return x + y
+
+with Flow("my handled flow!"):
+   first_result = add(1, y=2)
+   # or send as a keyword argument at task initialization
+   second_result = add(x=first_result, y=100, result_handler=LocalResultHandler())
+```
+
+## Choosing your result handlers
+
+In the above examples, we only used the `LocalResultHandler` class. This is one of several result handlers that integrate with different storage backends; the full list is in the API docs for [prefect.engine.results_handler](../../api/latest/engine/result_handlers.html) and more details on this interface is described in the concept ["Results and Result Handlers"](../concepts/results.md) documentation.
+
+We can write our own result handlers as long as they extend the [`ResultHandler`](https://github.com/PrefectHQ/prefect/blob/master/src/prefect/engine/result_handlers/result_handler.py) interface, or we can pick an existing implementation from Prefect Core that utilizes a storage backend we like; for example, I will use `prefect.engine.results_handler.GCSResultHandler` so that my data will be persisted in Google Cloud Storage.
+
+## Running a flow with `GCSResultHandler`
+
+Since the `GCSResultHandler` object must be instantiated with some initialization arguments, we utilize a flow-level override to pass in the Python object after configuring it:
+
+```python
+# flow.py
+from prefect.engine.result_handlers import GCSResultHandler
+
+gcs_handler = GCSResultHandler(bucket='prefect_results')
+
+with Flow("my handled flow!", result_handler=gcs_handler):
+   ...
+```
+
+Make sure your Prefect installation can [authenticate to Google's Cloud API](https://cloud.google.com/docs/authentication/getting-started). As long as the host of my Prefect installation can authenticate to this GCS bucket, each task's return value will be serialized into its own file in this GCS bucket.
+
+After running my flow, I can see that my task states know the key name of their individual result storage in `Result.safe_value` when I inspect them in-memory:
+
+```python
+>>> state = flow.run()
+>>> state.result[first_result]._result.safe_value
+<SafeResult: '2020/2/24/21120e41-339f-4f7d-a209-82fb3b1b5507.prefect_result'>
+>>> state.result[second_result]._result.safe_value
+<SafeResult: '2020/2/24/133eaf17-ab77-4468-afdf-734b6540dde0.prefect_result'>
+```
+
+Using [gsutil](https://cloud.google.com/storage/docs/gsutil), I can see that those keys exist in the GCS bucket I configured my results handler to submit data to:
+
+```bash
+$ gsutil ls -r gs://prefect_results
+gs://prefect_results/2020/:
+
+gs://prefect_results/2020/2/:
+
+gs://prefect_results/2020/2/24/:
+gs://prefect_results/2020/2/24/59082506-9217-436f-9d69-9ff569b20b7a.prefect_result
+gs://prefect_results/2020/2/24/a07dd6c1-837d-4925-be46-3b525be57779.prefect_result
+```
+
+If you are using Prefect Cloud, you can see that this metadata from the "safe value" is also stored by the scheduler and exposed in the UI:
```

Comment:
Strictly speaking the scheduler isn't involved - 
```suggestion
If you are using Prefect Cloud, you can see that this metadata from the "safe value" is also stored and exposed in the UI:
```

Reply:
Resolved in https://github.com/PrefectHQ/prefect/pull/2076/commits/b71055a4d1aaf889430c3659fab20044b9ddd068